go/mysql/mysql56_gtid_set.go

```diff
@@ -334,6 +334,83 @@ func (set Mysql56GTIDSet) AddGTID(gtid GTID) GTIDSet {
 	return newSet
 }
 
+// Union implements GTIDSet.Union().
+func (set Mysql56GTIDSet) Union(other GTIDSet) GTIDSet {
+	if set == nil || other == nil {
+		return set
+	}
+	mydbOther, ok := other.(Mysql56GTIDSet)
+	if !ok {
+		return set
+	}
+
+	// Make a copy and add the new GTID in the proper place.
+	// This function is not supposed to modify the original set.
+	newSet := make(Mysql56GTIDSet)
+
+	for OtherSID, otherIntervals := range mydbOther {
+		intervals, ok := set[OtherSID]
+
+		if !ok {
```

Comment:
Like `err`, the conventional `ok` var is an often-reused name, so I prefer to group the check with the assignment to reduce the chances of accidentally inserting other code in the middle.

```suggestion
		if !ok {
```

Reply:
Are you just saying to remove the extra space so it's physically closer to the assignment?

Reply:
Yeah.

Reply:
Removed the extra line.